lib/OpenQA/WebAPI/Controller/API/V1/Iso.pm

```diff
@@ -336,32 +343,49 @@ sub job_create_dependencies {
         my ($depname, $deptype) = @$dependency;
         next unless defined $settings->{$depname};
         for my $testsuite (_parse_dep_variable($settings->{$depname}, $settings)) {
-            if (!defined $testsuite_mapping->{$testsuite}) {
-                my $error_msg = "$depname=$testsuite not found - check for typos and dependency cycles";
+            my ($test, $machine) = @$testsuite;
+            for my $m (keys %{$testsuite_mapping->{$test}}) {
+                if (!exists $cluster_parents->{$test . ':' . $m}) {
+                    $cluster_parents->{$test . ':' . $m} = $testsuite_mapping->{$test}->{$m};
+                }
+            }
+            if (!defined $testsuite_mapping->{$test}->{$machine}) {
+                my $error_msg = "$depname=$test:$machine not found - check for dependency typos and dependency cycles";
                 OpenQA::Utils::log_warning($error_msg);
                 push(@error_messages, $error_msg);
             }
             else {
-                for my $parent (@{$testsuite_mapping->{$testsuite}}) {
-                    try {
-                        _check_for_cycle($job->id, $parent, $created_jobs);
-                    }
-                    catch {
-                        die "There is a cycle in the dependencies of $settings->{TEST}";
-                    };
-                    $self->db->resultset('JobDependencies')->create(
-                        {
-                            child_job_id  => $job->id,
-                            parent_job_id => $parent,
-                            dependency    => $deptype,
-                        });
-                }
+                my @parents = @{$testsuite_mapping->{$test}->{$machine}};
+                my %testsuite_info;
+                $testsuite_info{deptype}   = $deptype;
+                $testsuite_info{childtest} = $settings->{TEST};
+                $self->_create_dependencies($job, $created_jobs, \@parents, \%testsuite_info);
```

Comment:
Maybe passing `$deptype` and take name from `$job->settings_hash->{TEST}`

Reply:
you are right. it was redundant